Added instructions in faq.rst for insuring files are rendered as chapters in LaTeX

philip-schrodt opened the pull request:
This problem had been raised in a couple of earlier issues: the solution is straightforward but it took me quite a bit of experimentation before I figured it out, and in the meantime the LaTeX version looks rather ugly.

jakobandersen committed:
C++ initial refactoring to support templates.

Create symbol tables to represent all declarations, including templated
declarations (sphinx-doc/sphinx#1729).
Fixes sphinx-doc/sphinx#1314.

birkenfeld committed:
Merge branch 'stable'

birkenfeld committed:
Merge branch 'stable'

larsoner committed:
FIX: Only check if members is True

larsoner committed:
FIX: Ensure list is sent to OptParse

kieranh5511 committed:
Use xcolor over color where available

The boxes drawn around verbatim code (fcolorbox) with the color package seem to display poorly on-screen in some PDF readers (e.g. Adobe Reader). Boxes drawn with xcolor display much better, hence the xcolor package will be used if available.

RobRuana committed:
Closes #1904: [Napoleon] parses restructuredtext references in fields/params BEFORE splitting on colon

bgeron committed:
Fix a typo in doc/markup/toctree.rst.

jakobandersen committed:
C++, fixes sphinx-doc/sphinx#1913

Commit:
allow apidoc using 'python -m sphinx.apidoc'

Commit:
fix pep8 noise

shimizukawa committed:
fix flake8 issue

ericholscher committed:
Add suffix to HTML context. Fixes #1873

shimizukawa committed:
Merge pull request #1876 from ericholscher/source_suffix_in_build

Add suffix to doc context. Fixes #1873.

shimizukawa committed:
Merge pull request #1910 from bgeron/patch-1

Fix a typo in doc/markup/toctree.rst.

RobRuana committed:
Closed #1925: [Napoleon] provides @property examples & recommends PEP 484 type hints in Napoleon docs

RobRuana committed:
Closes #1927: [Napoleon] fixes parsing & formatting of types in Raises section

alex committed:
Use an HTTPS url for fonts so these load on HTTPS pages

Right now they're blocked as active mixed content.

birkenfeld committed:
merge with stable

RobRuana committed:
Related to #1927: [Napoleon] always start description on same line as type in Raises section

RobRuana committed:
Merge pull request #1928 from alex/patch-1

Use an HTTPS url for fonts so these load on HTTPS pages

RobRuana committed:
[Napoleon] Updates docs to use single quoted strings & 4-space indentation

RobRuana committed:
[Napoleon] Miscellaneous formatting fixes to napoleon example docstrings

jpic committed:
Link to Python's Style Guide for documentation.

This is a follow-up to #1931

lehmannro committed:
Explicitly mention sphinx-users mailing list

This should now answer searches for "mailing list" and "sphinx-", in case users just skim the Web site.

I have checked this in a Web browser and it looks good IMO (box not overfull.)

shimizukawa committed:
Merge pull request #1941 from sphinx-doc/oops-groups

Explicitly mention sphinx-users mailing list

warsaw committed:
One way to work around the lack of html.parser.HTMLParserError in Python 3.5

shimizukawa committed:
Merge pull request #1892 from Eric89GXL/fix-no-members

FIX: Only check if members is True. closes #1822, refs #1891, #1828, #1061, #1663

jwilk committed:
Fix typos.

shimizukawa committed:
Merge pull request #1951 from jwilk/spelling

Fix typos.

mitya57 committed:
Add support for SOURCE_DATE_EPOCH environment variable to ustrftime function

See https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal.

mitya57 committed:
Make epub builder use ustrftime instead of plain strftime

shimizukawa committed:
Merge pull request #1830 from thomasvandoren/ref-chapel-domain-in-docs

Add reference to Chapel domain in domains document.

hamogu committed:
Correct minor typo in confic.py

birkenfeld committed:
Merge pull request #1957 from hamogu/patch-1

Correct minor typo in confic.py

enhao committed:
Update zh_TW.py

birkenfeld committed:
Closes #1962: when adding directives, roles or nodes from an extension, warn if such
an element is already present (built-in or added by another extension).

Commit:
doc: adjust underline length of section headers

birkenfeld committed:
Merge pull request #1963 from myokoym/adjust-underline

doc: adjust underline length of section headers

birkenfeld committed:
fixup test suite for already-registered warnings

birkenfeld committed:
Closes #1960: show relative file path in output when copying dlfiles

birkenfeld committed:
Closes #1956: remove unused import in default conf.py

birkenfeld committed:
Merge pull request #1946 from warsaw/py35compat

One way to work around the lack of html.parser.HTMLParserError in Python 3.5

birkenfeld committed:
merge with stable

birkenfeld committed:
Merge pull request #1954 from mitya57/master

Add support for SOURCE_DATE_EPOCH environment variable

birkenfeld committed:
Merge pull request #1919 from pkienzle/master

allow apidoc using 'python -m sphinx.apidoc'

birkenfeld committed:
latex writer: fix questionable string escapes

birkenfeld committed:
Closes #1871: fix for latex writer with multirow and 1-column tables.

RobRuana committed:
Closes #1840: [Napoleon] Fixes Google style parsing to only match section headers ending with exactly one colon

kyleknap committed:
Update ``'releasename'`` latex element docs

To address https://github.com/sphinx-doc/sphinx/issues/1782

birkenfeld committed:
Fixes #1909: Add "doc" references to Intersphinx inventories.

birkenfeld committed:
Merge pull request #1966 from kyleknap/releasename-doc-update

Update ``'releasename'`` latex element docs

birkenfeld committed:
std domain: make sure that ref and doc references have proper classes set

eolo999 committed:
Document ``cmdoption`` directive deprecation

birkenfeld committed:
merge with stable

birkenfeld committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

birkenfeld committed:
Intersphinx: add a small helper to print out inventories when called from the command line with -m.

birkenfeld committed:
Fix test_build_html.

birkenfeld committed:
Merge pull request #1967 from eolo999/master

Document ``cmdoption`` directive deprecation

birkenfeld committed:
graphviz test: raise Skip if dot not available, instead of just passing

shimizukawa committed:
merge with stable

birkenfeld committed:
Merge branch 'stable'

birkenfeld committed:
erge branch 'stable'

birkenfeld committed:
Merge branch 'stable'

mtreinish committed:
Add support for converting checkmark unicode to latex symbol

This commit adds an entry to the list of tex escapes for the checkmark
unicode character. As part of this the use of the amssymb package is
added which is required for the checkmark symbol among others.

Closes #1959

birkenfeld committed:
Merge pull request #1971 from mtreinish/fix-checkmark

Add support for converting checkmark unicode to latex symbol

birkenfeld committed:
move amssymb package require to .sty file

vlasovskikh committed:
Closes #1968: Show extended type hints for function annotations that use 'typing' module

vlasovskikh committed:
Use 'SkipTest' from 'util' for tests

vlasovskikh committed:
Skip type hinting tests if there is a SyntaxError while importing a module with function annotations

birkenfeld committed:
Merge pull request #1975 from vlasovskikh/type-hints

Closes #1968: Show extended type hints for function annotations that use 'typing' module

pradyunsg committed:
Indent and Escape quote in Makefile

Small aesthetic changes that makes the Makefile look (and highlight) nicer.

pradyunsg committed:
Fix the quoting

Oops!

birkenfeld committed:
Merge pull request #2000 from pradyunsg/patch-1

Indent and Escape quote in Makefile

cjmayo committed:
Stop theme description stretching around screenshots in html output

progval committed:
Make grammar generation deterministic.

progval committed:
Make inventory generation deterministic.

progval committed:
Make Javascript locale deterministic.

progval committed:
Fix Python 2.6 incompatibility introduced in 4e77027.

daoiqi committed:
zh_TW.py apply  pep8 specification

jakobandersen committed:
C++, fix base class rendering.

progval committed:
Make searchindex generation deterministic.

jakobandersen committed:
C++, support for templates.

Fixes sphinx-doc/sphinx#1729.

jakobandersen committed:
C++, code cleanup.

jakobandersen committed:
C++, yet more code cleanup

jakobandersen committed:
C++, fix for template argument parsing.

Non-type template arguments where not parsed correctly. It is a bit
better now. The example in michaeljones/breathe#218 should work now.

jakobandersen committed:
C++, add namespace push/pop directives.

agjohnson committed:
Tune sphinx_rtd_theme version

With some recent changes that are slightly backwards compatible, we were going
to bump the version of the RTD theme. We wanted to drop the current version
pattern for the semver scheme, but Sphinx's requirement on the RTD theme is
pinned at >=0.1,<0.2. For now, we're stuck releasing on the current scheme.

This relaxes the versioning requirement for the theme, with 2.0 being the
following release that could contain backwards incompatible API changes. We'd
likely wait to bump to 1.0 until the next Sphinx release, as to not break
experience for users.

Refs snide/sphinx_rtd_theme#244

lehmannro committed:
Fixed #1786: Add configurable type hints.

This adds the option of giving, in addition to the type of the default
value, hints about permissible types for configuration values.

lehmannro committed:
Fixes #1984: Hint None being valid for primary_domain.

lehmannro committed:
Hint True/False being valid for autosummary_generate.

artakase committed:
Fix the language code for Portuguese

lehmannro committed:
Fix PEP8 complaint

pyflake8: E128 continuation line under-indented for visual indent.

jakobandersen committed:
Fix build

shimizukawa committed:
fix: unresolved any target.

shimizukawa committed:
avoid py3 syntax module

shimizukawa committed:
Merge pull request #2035 from artakase/master

Fix the language code for Portuguese in documentation

shimizukawa committed:
Merge branch 'stable'

shimizukawa committed:
Merge pull request #2031 from agjohnson/rtd-theme-version

Tune sphinx_rtd_theme version

shimizukawa committed:
Fix: "make check" fails on 'typing_test_data.py' on Py2 environment. Closes #2020.

shimizukawa committed:
Merge pull request #2009 from ProgVal/determinism

Make files generation at install deterministic

jakobandersen committed:
C++, fix parsing of signed/unsigned char.

jakobandersen committed:
C++, add missing support for 'friend' functions.

birkenfeld committed:
Port check_sources to single-source.

birkenfeld committed:
Rename "make check" to "make style-check"; "make check" is often used for test suites.

mhils committed:
Set napoleon_include_special_with_doc to False

Napoleon should respect Sphinx's default behaviour for documenting special members.

erosennin committed:
Omit empty URL targets in objects.inv

birkenfeld committed:
Merge pull request #2055 from erosennin/intersphinx-omit-empty-target

Fix: omit empty URL targets in objects.inv

Rapptz committed:
C++, add support for virtual base classes

Rapptz committed:
Fix flake8 error.

jakobandersen committed:
C++, document base class grammar, and cleanup.

jakobandersen committed:
C++, add support for final classes.

jakobandersen committed:
C++, add parameter pack expansion as base classes.

jakobandersen committed:
C++, fix parsing of types prefixed with 'enum'.

Fixes michaeljones/breathe#223.

jakobandersen committed:
C++, fix bug in type alias parsing.

Fixes sphinx-doc/sphinx#2061

TimKam committed:
Correct Dutch stop words

Before, Dutch stop word file was a not properly adjusted copy of the Danish stop word file

TimKam committed:
Update AUTHORS

TimKam committed:
Update CHANGES

birkenfeld committed:
Merge pull request #2063 from TimKam/patch-1

Bugfix: Dutch stop words were incorrect (Danish)

lehmannro committed:
Fix typo.

jakobandersen committed:
C++, remove symbol hax for template params

dean0x7d committed:
C++, add support for user-defined literals

jakobandersen committed:
C++, fix templated operator bug

Fixes sphinx-doc/sphinx#2058
Fixes sphinx-doc/sphinx#2080

jakobandersen committed:
Merge pull request #2081 from dean0x7d/cpp11literals

C++, add support for user-defined literals

jakobandersen committed:
C++, properly render multi-line declarations.

Fixes template rendering, e.g., when generating Latex.
Thanks to michaeljones/breathe#127 for pointing in the right direction.

jakobandersen committed:
C++, cross-reference update

(see also sphinx-doc/sphinx#2057)
- Elaborate the documentation in regard to templates.
- Add shorthand for template declarations.
- Add shorthand for (full) template specialisations.
- Add better error messages for unparseable references.

wilsoniya committed:
Intersphinx: inventory support for URLs w/ basic auth

vitaut committed:
Make sphinx_rtd_theme optional

Commit:
#2092: Add support for todo directive

Commit:
#2092: Update documentation

Commit:
Fixes #2092: Update authors and changelog

birkenfeld committed:
Merge pull request #2089 from wilsoniya/master

Intersphinx: inventory support for URLs w/ basic auth

birkenfeld committed:
intersphinx: style nits

shimizukawa committed:
fix typo

azazel75 committed:
Update Python3 grammar to Python3.5

Taken from https://hg.python.org/cpython/file/3.5/Grammar/Grammar@99014:0c704864b917

azazel75 committed:
Treat async/await as anonymous token and deal with them in tokenize.py

.. as cpython's parser does

azazel75 committed:
Add a test for doc on coroutine functions and general support for async/await py3.5 syntax

azazel75 committed:
Move py3.5 specific tests to their own file and adjust test environment

azazel75 committed:
Conditionally enable py3.5 tests in travis

Rapptz committed:
C++, Add support for template parameter info field lists

birkenfeld committed:
Merge pull request #2111 from azazel75/master

Upgrade pgen2 to support py3.5 async/await

lelit committed:
Include JS stemmers in the distribution

lelit committed:
Rectify the path of non-minified JS stemmers

RobRuana committed:
Merge pull request #2053 from mhils/patch-1

Set napoleon_include_special_with_doc to False by default

RobRuana committed:
Merge pull request #2093 from Sgiath/napoleon-todo

#2092: Napoleon 'todo' directive

hardtack committed:
Make sure to iterate obj_dict's keys using list, not view

In Python3 `dict.keys()` returns a view object not a list

birkenfeld committed:
Merge pull request #2116 from Hardtack/master

Make sure to iterate obj_dict's keys using list, not view

shimizukawa committed:
Merge pull request #2114 from lelit/master

Include JS stemmers in the distribution

shimizukawa committed:
Merge pull request #2033 from lehmannro/config-types-for-real

Allow hinting permissible types for configuration values

shimizukawa committed:
Merge pull request #2004 from cjmayo/theming

Stop theme description stretching around screenshots in html output

shimizukawa committed:
Merge branch 'stable'

RobRuana committed:
[Napoleon] Changes __unicode__ to conditionally use the "u" string literal to support Python 3.2 (which is still used by some users)

RobRuana committed:
Updates #2053 [Napoleon] Updates documentation to indicate napoleon_include_special_with_doc = True by default

RobRuana committed:
[Napoleon] Changes __unicode__ to use the six.u() function

mocobeta committed:
Fix performTermsSearch function for multiple words search.

mocobeta committed:
Remove old (current) codes.

mocobeta committed:
Replace tab characters with spaces.

mocobeta committed:
Fix alignments...

dean0x7d committed:
[Napoleon] Fix NumPy handling of *args and **kwargs parameters

RobRuana committed:
Merge pull request #2126 from dean0x7d/fix_numpy_kwargs

[Napoleon] Fix NumPy handling of *args and **kwargs parameters

shimizukawa committed:
Merge pull request #2121 from mocobeta/better_and_search

Fix performTermsSearch function for multiple words search.

shimizukawa committed:
merge with stable

shimizukawa committed:
Merge branch 'stable'

jakobandersen committed:
C++, tparam info field docs.

mocobeta committed:
MecabBinder should support both of Python 2 and 3.

jakobandersen committed:
Fix to parallel builds

Hax the C++ domain to support parallel builds when it's not used.
See sphinx-build/sphinx#2130.

jakobandersen committed:
C++, fix name lookup to match the current scope.

Fixes sphinx-doc/sphinx#2133.

jakobandersen committed:
C++, fix template parameter bug.

Symbols for template parameters should only have a declaration if the
parent symbol has.

shimizukawa committed:
Merge pull request #2127 from mocobeta/search_ja_mecab_py3

MecabBinder should support both of Python 2 and 3.

shimizukawa committed:
Merge with 'stable'

jakobandersen committed:
C++, full support for parallel builds.

Fixes sphinx-doc/sphinx#2130.

jakobandersen committed:
Merge branch 'master' of github.com:sphinx-doc/sphinx

jakobandersen committed:
Exclude stuff from style-check

shimizukawa committed:
Merge pull request #2141 from jakobandersen/style-check-fix

Exclude stuff from style-check

jakobandersen committed:
C++, don't remove symbols.

Symbols may be parents of some xref.

jakobandersen committed:
C++, fix sphinx-doc/sphinx#2145

Also fix lookup enumerators in unscoped enums, by their unscoped name.

jakobandersen committed:
C++, don't accidentally deepcopy all domain data.

See sphinx-doc/sphinx#2145.

shimizukawa committed:
merge with  'stable'

jakobandersen committed:
C++, add support for pointers to member (function)

See sphinx-doc/sphinx#2146

jakobandersen committed:
C++, fix line numbers in warnings.

See sphinx-doc/sphinx#2146.

shimizukawa committed:
separate changes into 1.4 and 1.3.x

shimizukawa committed:
merge with stable

shimizukawa committed:
Merge pull request #1757 from hemmecke/master

fix unclear documentation for html_last_updated_fmt

shimizukawa committed:
Update CHANGES refs #1757

shimizukawa committed:
Merge pull request #1804 from HassenPy/patch-1

sidebar text overflow

shimizukawa committed:
Update CHANGES refs #1804.

jakobandersen committed:
C++, parallel build fix

Delete operator symbols when clearing.

jakobandersen committed:
Fix flake8 error

daoiqi committed:
fix if user not set jieba options then os.path.isfile  error

daoiqi committed:
change zh_TW.py to zh.py

it also support Simplified Chinese Search

daoiqi committed:
add chinese search options into `html_search_options` document

jakobandersen committed:
Merge pull request #2163 from jakobandersen/master

C++, parallel build fix

Commit:
Fix #2177: Remove parallel hangs

usox committed:
perform a lower() on german index terms #2083

birkenfeld committed:
More parallel code optimizations.

birkenfeld committed:
merge stable to default

birkenfeld committed:
Merge pull request #2181 from usox/2083-fix_german_searchindex

perform a lower() on german index terms #2083

shimizukawa committed:
Merge with 'stable'

xuhdev committed:
Add the imgmath extension, the successor of pngmath.

In addition to the features which pngmath supports, imgmath also
supports svg output format.

tk0miya committed:
Fix #2113: Allow ``:class:`` option to code-block directive

xuhdev committed:
Update the pngmath doc to synchronize with imgmath

xuhdev committed:
Add Hong Xu to the AUTHOR list

xuhdev committed:
Replace pngmath with imgmath in quickstart.

xuhdev committed:
imgmath: Make the size of the displayed math customizable.

tk0miya committed:
Fix #2183 porterstemmer causes `make json` to fail

Drop PorterStemmer package support.
For now, PyStemmer can accelarate stemming process of snowballstemmer.

xuhdev committed:
Add deprecation information to sphinx.ext.pngmath

xuhdev committed:
LaTeX: Disable specific UTF-8 support commands for XeTeX and LuaTeX

XeTeX and LuaTeX have builtin UTF-8 support.

xuhdev committed:
Run xelatex and lualatex as well in latex tests.

jfbu committed:
Load graphicx without specifying the driver

This is issue #2164.

On the current main TeX installations (TeXLive, MikTeX), \pdftexversion is defined also when running latex to produce dvi files. This commit deletes TeX code which resulted in the "pdftex" driver being always assumed, which made it a priori impossible to compile via latex.

The graphicx packages is smart enough to detect correctly the driver (pdftex, xetex, ...) and defaults to dvips.

tk0miya committed:
Fix README

tk0miya committed:
Merge pull request #1899 from Eric89GXL/opts

FIX: Ensure list is sent to OptParse

tk0miya committed:
Upate CHANGES for PR #1899

QuLogic committed:
Sort attributes in inheritance diagrams.

This ensures that the graphviz script is always the same and thus the
filename (which is the hash of the script and other things) remains
consistent.

tk0miya committed:
Merge pull request #2192 from xuhdev/imgmath

Imgmath (pngmath with svg support)

tk0miya committed:
Update CHANGES for PR #2192

jwilk committed:
Fix typos.

tk0miya committed:
Merge pull request #2212 from jwilk/spelling

Fix typos

tk0miya committed:
Fix warnings caused by renaming pngmath in CHANGES and docs

tk0miya committed:
Update CHANGES for PR #2200

tk0miya committed:
Update CHANGES for PR #2204

tk0miya committed:
Add sphinx.ext.githubpages to publish the docs on GitHub Pages

xuhdev committed:
Specify a default format (pdf) in the LaTeX output Makefile

xuhdev committed:
Add "gz" and "xz" archive target to the LaTeX output Makefile

xuhdev committed:
The clean target the LaTeX ouput Makefile should remove more files

xuhdev committed:
Add more PHONY targets in the LaTeX output Makefile

tk0miya commented:
It seems this pull request has broken.
I could not find your commits.

Could you rebase this please?

tk0miya committed:
Merge pull request #1906 from kieranh5511/master

Use xcolor over color for \fcolorbox where available for LaTeX output

tk0miya committed:
Update CHANGES for #1906

uchida committed:
Fix #2165: Remove faulty and non-need conditional

tk0miya committed:
Update CHANGES for #2165

tk0miya committed:
Merge pull request #2216 from xuhdev/texinputs-Makefile

tk0miya committed:
Update CHANGES for PR #2216

birkenfeld committed:
Merge pull request #2210 from QuLogic/consistent-inheritance

Sort attributes in inheritance diagrams.

birkenfeld committed:
Merge pull request #2170 from daoiqi/zh-search

add Chinese Search into milestones

birkenfeld committed:
chinese search: add to changelog, remote porterstemmer support

tk0miya committed:
Merge pull request #2214 from tk0miya/2203_sphinx.ext.githubpages

Add sphinx.ext.githubpages to publish the docs on GitHub Pages

tk0miya committed:
Update CHANGES for PR #2214

vitaut committed:
Warn about sphinx_rtd_theme being unbundled

tk0miya committed:
Refactoring ``sphinx.ext.todo``; because of deprecation of make_admonition()

shimizukawa committed:
Merge with '87998df9cbef2380345d436121e6bca43345d2bd' on stable

Conflicts:
	tests/test_build_latex.py

tk0miya committed:
Merge branch 'stable'

tk0miya committed:
Merge pull request #1934 from jpic/styleguide_link

Link to Python's Style Guide for documentation.

shimizukawa committed:
merge with stable

birkenfeld committed:
Merge branch 'stable'

birkenfeld committed:
Obligatory copyright update.

shimizukawa committed:
Merge pull request #2087 from cppformat/optional-rtd

refs #2087, #2086, #1845, #2097: Make sphinx_rtd_theme optional

shimizukawa committed:
update CHANGES for #2087

xuhdev committed:
Move the testing code which spawns pdflatex to a function

There are currently two locations in the testing code running the same
code to spawn pdflatex. This commit puts these code into one single
function to reduce redundancy.

tk0miya committed:
Merge pull request #2236 from xuhdev/run-latex-tests

Move the testing code which spawns pdflatex to a function

tk0miya committed:
Update error message if sphinx_rtd_theme not found

tk0miya committed:
Fix #2231: Use DUrole instead of DUspan for custom roles in LaTeX writer

tk0miya commented:
Now, I close this PR.
If you want to merge your changes, please send new pull request.

Thanks,
